src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonEncodingStreamWrapper.cs

```diff
@@ -274,8 +274,7 @@ public override void WriteByte(byte b)
                 _stream.WriteByte(b);
                 return;
             }
-            _byteBuffer[0] = b;
-            Write(_byteBuffer, 0, 1);
+            Write(stackalloc byte[1] {b});
```

Comment:
This stream doesn't override Write(ReadOnlySpan).  As such, this is going to call the base Write(ReadOnlySpan) method, which will end up allocating a byte[1] per call.  That makes this change a regression rather than an improvement.

If you want to also refactor Write(byte[], int, int) into a method that calls to a new override of Write(ReadOnlySpan) with an appropriate implementation, then this can remain as you have it, otherwise it should be reverted.